config/initializers/rails_5_compatibility.rb

```diff
@@ -0,0 +1,23 @@
+# Rails 5.2 compatibility patches
+
+module SchemaStatementsRails5Compat
+  def foreign_key_exists?(source, target = nil, **options)
+    if Rails.version.start_with?('5.2')
```

Comment:
What if add do something like this first:
```ruby
      unless options.present?
        options = target
        Rails.logger.warn('migration incompatible with Rails 6, need to specify target table....')
      end
```

Reply:
We can throw warning, just unsure how useful it is :man_shrugging:

Reply:
might be useful for plugin maintainers if it's informative so they know what migrations they need to change and how

Reply:
Yeah, I meant they will totaly miss the warning in the migration log. Especially while they would have to run full migration (the old ones) to even have it in the logs. Wouldn't we have some better way to notify them maybe?